packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2782,6 +2782,60 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     }, /didInitAttrs called/);
   }
 
+  ['@test didInitAttrs is invoked before finishChains'](assert) {
+    expectDeprecation(/didInitAttrs called/);
+
+    let component;
+    let fooDidChangeCount = 0;
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: Component.extend({
+        init() {
+          this._super(...arguments);
+          component = this;
+        },
+
+        didInitAttrs() {
+          this.set('foo', 1);
+        },
+
+        fooDidChange: observer('foo', () => { fooDidChangeCount++; })
+      })
+    });
+
+    assert.equal(fooDidChangeCount, 0);
+    this.render('{{foo-bar}}');
+    assert.equal(fooDidChangeCount, 0);
+    run(() => component.set('foo', 2));
+    assert.equal(fooDidChangeCount, 1);
+  }
+
+  ['@test first willReceieveAttrs is invoked before finishChains'](assert) {
+    let component;
+    let fooDidChangeCount = 0;
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: Component.extend({
+        init() {
+          this._super(...arguments);
+          component = this;
+        },
+
+        willReceieveAttrs() {
+          this.set('foo', 1);
+        },
+
+        fooDidChange: observer('foo', () => { fooDidChangeCount++; })
+      })
+    });
+
+    assert.equal(fooDidChangeCount, 0);
+    this.render('{{foo-bar}}');
+    assert.equal(fooDidChangeCount, 0);
+    run(() => component.set('foo', 2));
+    assert.equal(fooDidChangeCount, 1);
+  }
+
```

Comment:
Just to keep track of the implicit requirements, to make things easier when revisiting in the future